### ISSUE TITLE
Unable to resolve dependencies due sdk restrictions

### ISSUE THREAD

**Issue by gbloggs** [dart-ogurets/openapi_dart_common#10]
We have our pubspec.yaml set to:
```
...
environment:
  sdk: '^3.0.0'
...
```
attempting to do a pub get gives:
```
W:\tools\flutter3\bin\cache\dart-sdk\bin\dart.exe pub get
Resolving dependencies...
The current Dart SDK version is 3.1.4.

Because wibble depends on openapi_dart_common >=5.1.0 which requires SDK version >=2.15.0 <3.1.0, version solving failed.
```

this is also noted on pub.dev under the static analysis:
```
0/50 points: code has no errors, warnings, lints, or formatting issues

    Running dart pub outdated failed with the following output:

The current Dart SDK version is 3.2.0-210.1.beta.
Because openapi_dart_common requires SDK version >=2.15.0 <3.1.0, version solving failed.
```

**Comment by rvowles** [comment]
Fascinating, I'm just not keeping up with these Dart versions, I will pull a container I think with this set of latest versions and release a new version. "wibble"?

**Comment by rvowles** [comment]
OK - released 5.1.1 which has an upper bound of 3.99.0 - and it should be picked up by the >= 5.1.0 you are using.
